crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -700,14 +700,118 @@ impl Analyzer<'_, '_> {
 
         if !self.rule().always_strict && types.len() == 2 {
             let (a, b) = (&types[0], &types[1]);
-
-            if (a.is_str_lit() && b.is_str_lit() || (a.is_num_lit() && b.is_num_lit()) || (a.is_bool_lit() && b.is_bool_lit()))
+            if ((a.is_str_lit() && b.is_str_lit()) || (a.is_num_lit() && b.is_num_lit()) || (a.is_bool_lit() && b.is_bool_lit()))
                 && !a.type_eq(b)
             {
                 return never!();
             }
         }
 
+        let enum_variant_iter = types.iter().filter(|&t| t.is_enum_variant()).collect::<Vec<&Type>>();
+        let enum_variant_len = enum_variant_iter.len();
+
+        if enum_variant_len > 0 {
+            if let Some(first_enum) = enum_variant_iter.first() {
+                let mut enum_temp = first_enum.normalize();
+                let mut is_enum_value_nq = false;
+                for elem in enum_variant_iter.into_iter() {
+                    if let Type::EnumVariant(el) = elem.normalize() {
+                        if let Type::EnumVariant(en) = enum_temp {
+                            if let Type::EnumVariant(EnumVariant { name: None, .. }) = enum_temp {
+                                enum_temp = elem;
+                                continue;
+                            } else if en.enum_name != el.enum_name {
+                                return never!();
+                            } else {
+                                // eq two arguemnt enum_name
+                                if let Ok(el_lit) = self.expand_enum_variant(elem.clone()) {
+                                    if let Ok(etl) = self.expand_enum_variant(enum_temp.clone()) {
+                                        if !etl.type_eq(&el_lit) {
+                                            is_enum_value_nq = true;
```

Comment:
Can't we use `return never!()` here?

Reply:
I guess I was hasty. I agree.
Thank you for your feedback.